woocommerce/woocommerce-android
# Issue/6343 adjust to compose guidelines

Closes: #6343 

### Description
Adjust More Menu and Inbox screen to comply with compose guidelines: 
- Remove hard-coded dimens
- Add proper naming to files and root composable functions
- Remove unnecessary xml files

EXTRA: The clickable background/effect on More Menu header to open site switch screen was kind of weird. It did not match parent width. I've updated this to use a `Box` layout and extend the clickable area to the full width of the row.

### Testing instructions
Just navigate to More Menu and Inbox screen and double check there is nothing wrong with paddings/margins and sizes. 
Click on More Menu header and check the new clickable effect. 

### Images/gif

https://user-images.githubusercontent.com/2663464/165460703-c4a6a28b-f010-49e8-94ad-df05c90ac68c.mp4
